src/plugins/data_views/common/types.ts

```diff
@@ -111,50 +182,143 @@ export interface DataViewAttributes {
 export type IndexPatternAttributes = DataViewAttributes;
 
 /**
- * @intenal
- * Storage of field attributes. Necessary since the field list isn't saved.
+ * Set of field attributes
+ * @public
  */
 export interface FieldAttrs {
   [key: string]: FieldAttrSet;
 }
 
+/**
+ * Field attributes that are stored on the data view
+ * @public
+ */
 export interface FieldAttrSet {
+  /**
+   * Custom field label
+   */
   customLabel?: string;
+  /**
+   * Popularity count - used for discover
+   */
   count?: number;
 }
 
+/**
+ * Handler for data view notifications
+ * @public
+ * @param toastInputFields Toast notif config
+ * @param key Used to dedupe notifs
+ */
 export type OnNotification = (toastInputFields: ToastInputFields, key: string) => void;
+
+/**
+ * Handler for data view errors
+ * @public
+ * @param error Error object
+ * @param toastInputFields Toast notif config
+ * @param key Used to dedupe notifs
+ */
 export type OnError = (error: Error, toastInputFields: ErrorToastOptions, key: string) => void;
 
+/**
+ * interface for UiSettings common interface
```

Comment:
Do we want to avoid using, e.g., `{@link UiSettings}` for docs like these? I guess it does not do what we expect when it gets printed to docs site in HTML?

Reply:
Its a good suggestion which I added for this instance. I'm not going to go through this PR and add them but will keep it in mind in the future.